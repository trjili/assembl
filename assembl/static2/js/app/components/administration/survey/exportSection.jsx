import React from 'react';
import { I18n, Translate } from 'react-redux-i18n';
import { Link } from 'react-router';

import { getDiscussionId } from '../../../utils/globalFunctions';
import SectionTitle from '../sectionTitle';

const ExportSection = ({ i18n }) => {
  const debateId = getDiscussionId();
  return (
    <div className="admin-box">
      <SectionTitle i18n={i18n} phase="survey" tabId="2" annotation={I18n.t('administration.surveyExport.annotation')} />
      <div className="admin-content">
        <Link className="button-link button-dark margin-l" href={`/data/Discussion/${debateId}/phase1_csv_export`}>
          <Translate value="administration.surveyExport.link" />
        </Link>
      </div>
    </div>
  );
};

export default ExportSection;