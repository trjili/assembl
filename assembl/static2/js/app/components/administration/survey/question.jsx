import React from 'react';
import { gql, graphql, withApollo } from 'react-apollo';
import { Link } from 'react-router';
import { I18n } from 'react-redux-i18n';
import { Row, Col } from 'react-bootstrap';
import SectionTitle from '../sectionTitle';
import VideoForm from './videoForm';
import { getDiscussionSlug } from '../../../utils/globalFunctions';

const GetThematics = gql`
{
  thematics(identifier:"survey") {
    id,
    titleEntries {
      localeCode,
      value
    },
    video {
      htmlCode,
      title,
      description
    }
  }
}
`;

const getVideoByThematicId = (thematics, queriedId) => {
  let video = {};
  thematics.forEach((thematic) => {
    if (thematic.id === queriedId) {
      video = thematic.video;
    }
  });
  return video;
};

const Question = ({ client, data, i18n, selectedLocale, queriedId }) => {
  
  const thematicsData = client.readQuery({ query: GetThematics });
  const thematics = thematicsData.thematics || [];
  const slug = getDiscussionSlug();
  
  const video = getVideoByThematicId(thematics, queriedId);
  
  return (
    <div className="admin-box">
      <SectionTitle i18n={i18n} phase="survey" tabId="1" annotation={I18n.t('administration.annotation')} />
      <div className="admin-content">
        <Row>
          {thematics.map((thematic, index) => {
            return(
              <Col xs={12} md={Math.round(12 / thematics.length)} key={index}>
                <Link className="tab-title" activeClassName="tab-title-active" to={`/${slug}/administration/survey?section=2&thematic=${thematic.id}`}>
                  {`${I18n.t('administration.thematic')} ${index + 1}`}
                </Link>
              </Col>
            )
          })}
        </Row>
        <Row className="margin-xl">
          <VideoForm video={video} selectedLocale={selectedLocale} id={queriedId} />
        </Row>
      </div>
    </div>
  );
};

export default withApollo(graphql(GetThematics)(Question));