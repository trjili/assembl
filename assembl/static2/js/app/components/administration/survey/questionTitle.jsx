import React from 'react';
import { gql, withApollo } from 'react-apollo';
import { FormControl } from 'react-bootstrap';

const GetThematics = gql`
{
  thematics(identifier:"survey") {
    id,
    titleEntries {
      localeCode,
      value
    },
    questions {
      titleEntries {
        localeCode,
        value
      }
    }
  }
}
`;

const updateTitle = (client, tIndex, qIndex, selectedLocale, titleEntryIndex, value) => {
  const thematicIndex = tIndex;
  const questionIndex = qIndex;
  const entryIndex = titleEntryIndex;

  const thematicsData = client.readQuery({ query: GetThematics });
  const thematics = thematicsData.thematics;

  const newTitleEntries = {
    localeCode: selectedLocale,
    value: value,
    __typename: 'LangStringEntry'
  };

  if (entryIndex === -1) {
    thematics[thematicIndex].questions[questionIndex].titleEntries.push(newTitleEntries);
  } else {
    thematics[thematicIndex].questions[questionIndex].titleEntries.splice(entryIndex, 1, newTitleEntries);
  }

  client.writeQuery({
    query: GetThematics,
    data: thematicsData
  });
};

const QuestionsTitle = ({ client, tIndex, qIndex, titleEntries, selectedLocale }) => {
  const titleEntry = titleEntries.find((entry) => {
    return entry.localeCode === selectedLocale;
  });
  const title = titleEntry ? titleEntry.value : '';
  const titleEntryIndex = titleEntries.indexOf(titleEntry);

  const handleQuestionChange = (e) => {
    updateTitle(client, tIndex, qIndex, selectedLocale, titleEntryIndex, e.target.value);
  };

  return (
    <FormControl
      componentClass="textarea"
      className="text-area"
      value={title}
      onChange={handleQuestionChange}
    />
  );
};

export default withApollo(QuestionsTitle);