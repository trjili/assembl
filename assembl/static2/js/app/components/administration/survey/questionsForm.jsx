import React from 'react';
import { gql, withApollo } from 'react-apollo';
import { FormGroup } from 'react-bootstrap';

import QuestionTitle from './questionTitle';

const GetThematics = gql`
{
  thematics(identifier:"survey") {
    id,
    titleEntries {
      localeCode,
      value
    },
    questions {
      titleEntries {
        localeCode,
        value
      }
    }
  }
}
`;

const QuestionsForm = ({ client, thematicId, lang }) => {
  const thematicsData = client.readQuery({ query: GetThematics });

  const thematics = thematicsData.thematics;
  const thematic = thematics.find((t) => {
    return String(t.id) === thematicId;
  });

  const thematicIndex = thematics.findIndex((t) => {
    return String(t.id) === thematicId;
  });

  const questions = thematic ? thematic.questions : [];

  const addQuestion = () => {
    thematicsData.thematics[thematicIndex].questions.push({
      titleEntries: [],
      __typename: 'Question'
    });
    client.writeQuery({
      query: GetThematics,
      data: thematicsData
    });
  };

  return (
    <div className="form-container">
      <div className="margin-xl">
        {questions.map((question, index) => {
          return (
            <FormGroup key={index}>
              <QuestionTitle tIndex={thematicIndex} qIndex={index} titleEntries={question.titleEntries} selectedLocale={lang} />
            </FormGroup>
          );
        })}
        <div onClick={addQuestion} className="plus margin-l">+</div>
      </div>
    </div>
  );
};

export default withApollo(QuestionsForm);