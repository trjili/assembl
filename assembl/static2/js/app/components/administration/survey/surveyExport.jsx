import React from 'react';

class surveyExport extends React.Component {
  render() {
    const { showSection } = this.props;
    return (
      <p className={showSection ? 'shown' : 'hidden'}>surveyExport assembl</p>
    );
  }
}

export default surveyExport;