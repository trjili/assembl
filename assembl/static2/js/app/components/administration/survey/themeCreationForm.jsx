import React from 'react';
import { connect } from 'react-redux';
import { Translate, I18n } from 'react-redux-i18n';
import { FormGroup, FormControl } from 'react-bootstrap';

import { updateThemeTitle, updateThemeImage } from '../../../actions/adminActions';
import ImageUploader from '../../common/imageUploader';

const ThemeCreationForm = ({ id, image, selectedLocale, title, updateTitle, updateImage }) => {
  const trsl = I18n.t('administration.ph.title');
  const ph = `${trsl} ${selectedLocale.toUpperCase()}`;
  const index = (Number(id) + 1).toString();
  return (
    <div className="form-container">
      <div className="title">
        <Translate value="administration.themeNum" index={index} />
      </div>
      <FormGroup>
        <FormControl
          type="text"
          placeholder={ph}
          value={title}
          onChange={(e) => {
            return updateTitle(e.target.value);
          }}
        />
      </FormGroup>
      <FormGroup>
        <ImageUploader
          file={image}
          handleChange={(e) => {
            return updateImage(e.target.files);
          }}
        />
      </FormGroup>
      <div className="pointer right">
        <span className="assembl-icon-delete grey" />
      </div>
      <div className="separator" />
    </div>
  );
};

ThemeCreationForm.defaultProps = {
  title: ''
};

const mapStateToProps = ({ admin }, { id, selectedLocale }) => {
  const theme = admin.surveyThemesById[id];
  return {
    image: theme.image,
    title: theme.titlesByLocale[selectedLocale]
  };
};

const mapDispatchToProps = (dispatch, { id, selectedLocale }) => {
  return {
    updateTitle: (value) => {
      return dispatch(updateThemeTitle(id, selectedLocale, value));
    },
    updateImage: (files) => {
      return dispatch(updateThemeImage(id, files[0]));
    }
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(ThemeCreationForm);