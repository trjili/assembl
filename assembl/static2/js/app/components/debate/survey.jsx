import React from 'react';
import { connect } from 'react-redux';
import { graphql } from 'react-apollo';
import { Translate } from 'react-redux-i18n';
import gql from 'graphql-tag';
import { Grid, Row, Col } from 'react-bootstrap';
import Loader from '../common/loader';
import ThematicPreview from '../common/thematicPreview';

class Survey extends React.Component {
  render() {
    const { ideas, loading } = this.props.data;
    return (
      <section className="themes-section">
        {loading && <Loader color="black" />}
        {ideas &&
          <Grid fluid className="background-grey">
            <div className="max-container">
              <div className="title-section">
                <div className="title-hyphen">&nbsp;</div>
                <h1 className="dark-title-1">
                  <Translate value="debate.survey.themesTitle" />
                </h1>
              </div>
              <div className="content-section">
                <Row className="no-margin">
                  {ideas.map((idea, index) => {
                    return(
                      <div key={`thematic-${index}`} >
                        {idea.title &&
                          <Col xs={12} sm={6} md={3} className="no-padding">
                            <ThematicPreview bkgImgUrl={'/data/Discussion/6/documents/'+(421+index)+'/data'} nbPosts={idea.numPosts} nbContributors="12" link="" title={idea.title} description={idea.description} />
                          </Col>
                        }
                      </div>
                    )
                  })}
                </Row>
              </div>
            </div>
          </Grid>
        }
      </section>
    );
  }
}

Survey.propTypes = {
  data: React.PropTypes.shape({
    loading: React.PropTypes.bool.isRequired,
    error: React.PropTypes.object,
    thematics: React.PropTypes.object,
  }).isRequired,
};

const ThematicQuery = gql`
  query ThematicQuery($lang: String!) {
   ideas {
     ... on Thematic {
       id,
       title(lang: $lang),
       description,
       numPosts
     }
   }
  }
`;

const SurveyWithData = graphql(ThematicQuery, {
  options: ({ i18n }) => ({
    variables: {
      lang: i18n.locale
    }
  }),
})(Survey);

const mapStateToProps = (state) => {
  return {
    i18n: state.i18n
  };
};

export default connect(mapStateToProps)(SurveyWithData);