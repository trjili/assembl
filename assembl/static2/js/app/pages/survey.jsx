import React from 'react';
// import { connect } from 'react-redux';
import { connect, graphql } from 'react-apollo';
import update from 'immutability-helper';
import gql from 'graphql-tag';
import { I18n, Translate } from 'react-redux-i18n';
import { Grid, Button } from 'react-bootstrap';
import Alert from '../components/common/alert';
import Modal from '../components/common/modal';
import Loader from '../components/common/loader';
import Video from '../components/debate/survey/video';
import Header from '../components/debate/survey/header';
import Question from '../components/debate/survey/question';
import Navigation from '../components/debate/survey/navigation';
import Proposals from '../components/debate/survey/proposals';

class Survey extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      moreProposals: false,
      showModal: false
    };
    this.showMoreProposals = this.showMoreProposals.bind(this);
    this.getIfProposals = this.getIfProposals.bind(this);
    this.redirectToLogin = this.redirectToLogin.bind(this);
    this.displayAlert = this.displayAlert.bind(this);
  }
  getIfProposals(questions) {
    this.questions = questions;
    if (!this.questions) return false;
    let isProposals = false;
    this.questions.forEach((question) => {
      if (question.posts.edges.length > 0) isProposals = true;
    });
    return isProposals;
  }
  showMoreProposals() {
    this.setState({
      moreProposals: true
    });
  }
  redirectToLogin() {
    const { connectedUserId } = this.props.context;
    if (connectedUserId) {
      this.setState({
        showModal: false
      });
    } else {
      this.setState({
        showModal: true
      });
    }
  }
  displayAlert(style, msg) {
    this.setState({
      alertStyle: style,
      alertMsg: msg,
      showAlert: true
    });
    setTimeout(() => {
      this.setState({
        showAlert: false
      });
    }, 10000);
  }
  render() {
    console.log(this.props);
    const { loading, theme } = this.props.data;
    const { rootPath } = this.props.context;
    const { debateData } = this.props.debate;
    return (
      <div className="survey">
        {loading && <Loader color="black" />}
        {theme &&
          <div className="relative">
            <Header title={theme.title} imgUrl={theme.imgUrl} />
            <Modal
              body={I18n.t('debate.survey.modalBody')}
              link={`${rootPath}${debateData.slug}/login`}
              footer={I18n.t('debate.survey.modalFooter')}
              showModal={this.state.showModal}
            />
            <Alert showAlert={this.state.showAlert} style={this.state.alertStyle} msg={this.state.alertMsg} />
            {theme.video &&
              <Video
                title={theme.video.title}
                description={theme.video.description}
                htmlCode={theme.video.htmlCode}
              />
            }
            <div className="questions">
              {theme.questions && theme.questions.map((question, index) => {
                return (
                  <Question
                    redirectToLogin={this.redirectToLogin}
                    displayAlert={this.displayAlert}
                    title={question.title}
                    index={index + 1}
                    key={index}
                    questionId={question.id}
                  />
                );
              })}
            </div>
            {theme.questions &&
              <Navigation questionsLength={theme.questions.length} />
            }
            <div className="proposals">
              <section className="proposals-section" id="proposals">
                <Grid fluid className="background-light">
                  <div className="max-container">
                    <div className="question-title">
                      <div className="title-hyphen">&nbsp;</div>
                      <h1 className="dark-title-1">
                        <Translate value="debate.survey.proposalsTitle" />
                      </h1>
                    </div>
                    <div className="center">
                      {theme.questions && theme.questions.map((question, index) => {
                        return (
                          <Proposals
                            title={question.title}
                            posts={question.posts.edges}
                            moreProposals={this.state.moreProposals}
                            questionIndex={index + 1}
                            redirectToLogin={this.redirectToLogin}
                            displayAlert={this.displayAlert}
                            key={index}
                          />
                        );
                      })}
                      {(!this.state.moreProposals && this.getIfProposals(theme.questions)) &&
                        <Button className="button-submit button-dark" onClick={this.showMoreProposals}>
                          <Translate value="debate.survey.moreProposals" />
                        </Button>
                      }
                    </div>
                    <div className="margin-xl">&nbsp;</div>
                  </div>
                </Grid>
              </section>
            </div>
          </div>
        }
      </div>
    );
  }
}

const ThemeQuery = gql`
  query ThemeQuery($lang: String!, $id: ID!) {
    theme: node(id: $id) {
      ... on Thematic {
        title(lang: $lang),
        imgUrl,
        id,
        video(lang: $lang){
          title,
          description,
          htmlCode
        }
        questions {
          ... on Question {
            title(lang: $lang),
            id,
            posts{
              edges {
                node {
                  ... on PropositionPost {
                    id,
                    body,
                    mySentiment,
                    sentimentCounts {
                      like,
                      disagree
                    }
                  }
                }
              }
            }
          }
        }
      }
    }
  }
`;

// Survey.propTypes = {
//   data: React.PropTypes.shape({
//     loading: React.PropTypes.bool.isRequired,
//     error: React.PropTypes.object,
//     theme: React.PropTypes.Array
//   }).isRequired
// };

// const SurveyWithData = graphql(ThemeQuery, {
//   props: ({ ownProps, data }) => ({
//     data: data
//   }),
//   options({ params }) {
//     return {
//       reducer: (previousResult, action, variables) => {
//         if (action.type === 'APOLLO_MUTATION_RESULT' && action.operationName === 'createPost'){
//           return update(previousResult, {
//             theme: {
//               questions: {
//                 0: {
//                   posts: {
//                     edges: {
//                       $unshift: [{ node: action.result.data.createPost.post }]
//                     }
//                   }
//                 }
//               }
//             }
//           });
//         }
//         return previousResult;
//       },
//     };
//   }
// })(Survey);

const mapStateToProps = (state) => {
  return {
    lang: state.i18n.locale,
    context: state.context,
    debate: state.debate
  };
};

const mapQueriesToProps = (state) => {
  return {
    data: { ThemeQuery }
  }
}

// export default connect(mapStateToProps)(SurveyWithData);


export default connect(mapQueriesToProps)(Survey);
