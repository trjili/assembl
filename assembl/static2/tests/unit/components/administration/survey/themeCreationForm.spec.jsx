import React from 'react';
import renderer from 'react-test-renderer';

import { ThemeCreationForm } from '../../../../../js/app/components/administration/survey/themeCreationForm';

describe('ThemeCreationForm component', () => {
  it('should render a theme creation form', () => {
    const myFile = new File([''], 'foobar.png');
    const removeSpy = jest.fn(() => {});
    const updateTitleSpy = jest.fn(() => {});
    const updateImageSpy = jest.fn(() => {});
    const props = {
      id: '0',
      image: myFile,
      remove: removeSpy,
      selectedLocale: 'fr',
      title: 'My theme',
      updateTitle: updateTitleSpy,
      updateImage: updateImageSpy
    };
    const component = renderer.create(<ThemeCreationForm {...props} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});