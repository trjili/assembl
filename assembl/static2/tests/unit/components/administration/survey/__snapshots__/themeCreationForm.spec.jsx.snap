exports[`ThemeCreationForm component should render a theme creation form 1`] = `
<div
  className="form-container">
  <div
    className="title">
    <span>
      Theme Num
    </span>
  </div>
  <div
    className="form-group">
    <input
      className="form-control"
      id={undefined}
      onChange={[Function]}
      placeholder="Title FR"
      type="text"
      value="My theme" />
  </div>
  <div
    className="form-group">
    <div>
      <div>
        <span>
          foobar.png
        </span>
        <img
          alt="foobar.png"
          className="img-thumbnail"
          src="" />
      </div>
      <button
        className="btn btn-default"
        disabled={false}
        onClick={[Function]}
        type="button">
        <span>
          Upload Button
        </span>
      </button>
      <input
        accept="image/*"
        onChange={[Function]}
        style={
          Object {
            "display": "none",
          }
        }
        type="file" />
    </div>
  </div>
  <div
    className="pointer right">
    <button
      className="btn btn-default"
      disabled={false}
      onClick={[Function]}
      type="button">
      <span
        className="assembl-icon-delete grey" />
    </button>
  </div>
  <div
    className="separator" />
</div>
`;
